objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java

```diff
@@ -166,6 +169,114 @@ public void testImageSaving() throws ImageProcessingException {
+            fail("Could not setup the JSON YoloProcessor");
+        }
+        
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+            
+            NeuralNetResults results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+            
+            // Should save first image with timestamp
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 1;
+            assert d.listFiles()[0].getName().matches(timestampPattern);
+            
+            results = null;
+            results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+            
+            // Every other so second should not save
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 1;
+            
+            results = null;
+            results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+            
+            // Every other so third and first should be saved
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 2;
+            assert d.listFiles()[0].getName().matches(timestampPattern);
+            assert d.listFiles()[1].getName().matches(timestampPattern);
+            
+        } finally {
+            // delete the directory even if the test fails
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+            
+            ypImageSaver.close();
+        }
+    }
+    
+    // Similar to testImageSaving() and testImageSaving2(), but saveImage is set to "vantiq".
+    // No images should be saved locally.
+    @Test
+    public void testImageSavingVantiq() throws ImageProcessingException {
+        Map config = new LinkedHashMap<>();
+        YoloProcessor ypImageSaver = new YoloProcessor();
+        
+        config.put("pbFile", PB_FILE);
+        config.put("labelFile", LABEL_FILE);
+        config.put("outputDir", OUTPUT_DIR);
+        config.put("saveRate", SAVE_RATE);
+        config.put("saveImage", "vantiq");
+        try {
+            ypImageSaver.setupImageProcessing(config, MODEL_DIRECTORY, AUTH_TOKEN, SERVER);
+        } catch (Exception e) {
+            fail("Could not setup the JSON YoloProcessor");
+        }
+        
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+            
+            NeuralNetResults results = ypImageSaver.processImage(getTestImage());
```

Comment:
So, does this mean that we saved something to VANTIQ but we 1) don't test that it got there, 2) don't check that it's named correctly, and 3) don't delete it?

We need to enhance the tests to verify that things arrived, are correct, and when we're done we wipe them out.  This is also a security issue since we've now made a few auth token's visible that we should not.

If other's (like Brett, etc.) run the tests, they'll fill up your system. This is another reason why we should extract the server & authtoken & have them be properties that someone adds to gradle.  Then, your namespace isn't relied on and doesn't get all the crap.  I just ran the tests (to verify) so that added some more images to your space.